cmd/snap/inhibit.go

```diff
@@ -34,143 +34,160 @@ import (
 	"github.com/snapcore/snapd/usersession/client"
 )
 
-func waitWhileInhibited(snapName string) error {
-	hint, _, err := runinhibit.IsLocked(snapName)
-	if err != nil {
-		return err
-	}
-	if hint == runinhibit.HintNotInhibited {
-		return nil
+var runinhibitWaitWhileInhibited = runinhibit.WaitWhileInhibited
+
+func waitWhileInhibited(snapName string) (err error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer func() {
+		if err != nil {
+			cancel()
```

Comment:
I don't think a think it is safe to `defer cancel()`, what if the underlying notification mechanism is async? I like `signal.NotifyContext` approach better.

Reply:
Actually I am bit confused by this, could please explain how we could cancel the context safely without accidentally stopping a notification that was run in a separate go routine midway?
After digging through the implementation I think we are safe here because non of those notification are async anyway.